rust/kernel/error.rs

```diff
@@ -58,6 +58,26 @@ impl Error {
     pub fn to_kernel_errno(&self) -> c_types::c_int {
         self.0
     }
+
+    /// Creates an [`Error`] from a kernel pointer.
+    ///
+    /// # Pointer value range
+    ///
+    /// According to `include/linux/err.h`:
+    /// - [0, .., `core::usize::MAX - bindings::MAX_ERRNO`) is the range for
+    ///   normal values of pointer.
+    ///
+    /// - [`core::usize::MAX - bindings::MAX_ERRNO`,..,`core::usize::MAX] is the
+    ///   range for error values stored in pointer.
+    pub fn from_kernel_ptr(ptr: *mut c_types::c_void) -> Option<Error> {
+        let value = ptr as usize;
+
+        if value >= core::usize::MAX - bindings::MAX_ERRNO as usize {
```

Comment:
Does this expose a kernel implementation detail (i.e. the way errors are encoded inside pointers)? Would it be more convenient to use the kernel function `IS_ERR()` here? We know it'll always work.

Reply:
I tend to agree. The `as usize` should both be fine (the first is from a pointer and the second could be put into a `const` variable, which I would do anyway).